assets/js/blocks/products/base-utils.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * External dependencies
+ */
+import { getRegisteredInnerBlocks } from '@woocommerce/blocks-registry';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ProductListTitle,
+	ProductListPrice,
+	ProductListButton,
+	ProductListImage,
+	ProductListRating,
+	ProductListSummary,
+	ProductListSaleBadge,
+} from '../../atomic/components/product-list';
+
+// @todo how to support these using mapping?
+//'core/paragraph',
+//'core/heading',
+/**
+ * Map blocks names to components.
+ *
+ * @param {string} blockName Name of the parent block. Used to get extension children.
+ */
+export const getBlockMap = ( blockName ) => ( {
+	'woocommerce/product-list-price': ProductListPrice,
+	'woocommerce/product-list-image': ProductListImage,
+	'woocommerce/product-list-title': ProductListTitle,
+	'woocommerce/product-list-rating': ProductListRating,
+	'woocommerce/product-list-button': ProductListButton,
+	'woocommerce/product-list-summary': ProductListSummary,
+	'woocommerce/product-list-sale-badge': ProductListSaleBadge,
+	...getRegisteredInnerBlocks( blockName ),
+} );
+
+/**
+ * Maps component names to component classes and block names.
+ *
+ * @param {string} blockName Name of the parent block. Used to get extension children.
+ * @param {object[]} blockMap Map of blocks as returned by `getBlockMap`. If not specified, it gets the block map with the passed `blockName`.
+ */
+export const getReversedBlockMap = (
+	blockName,
+	blockMap = getBlockMap( blockName )
+) =>
+	Object.entries( blockMap ).reduce( ( acc, block ) => {
+		acc[ block[ 1 ].name ] = {
+			component: block[ 1 ],
+			key: block[ 0 ],
+		};
+		return acc;
+	}, {} );
```

Comment:
What can we do to eliminate the necessity of this?  Could registration automatically add things to the map in this format?

Reply:
Yes... I didn't really like this part. I have spent some more time thinking on this and came with a refactor that removes the need of the reverse map: 2b1014097628e76fa7ea78e79b7e17d343d03dd0.